Sources/BraveWallet/Crypto/Stores/NFTStore.swift

```diff
@@ -62,6 +71,54 @@ public class NFTStore: ObservableObject {
     userAssetManager: self.assetManager
   )
   
+  enum NFTDisplayType: Int, CaseIterable, Identifiable {
+    case visible
+    case invisible
+    case spam
+    
+    var id: Int {
+      rawValue
+    }
+    
+    var dropdownTitle: String {
+      switch self {
+      case .visible:
+        return Strings.Wallet.nftsTitle
+      case .invisible:
+        return Strings.Wallet.nftHidden
+      case .spam:
+        return Strings.Wallet.nftSpam
+      }
+    }
+    
+    var emptyTitle: String {
+      switch self {
+      case .visible:
+        return Strings.Wallet.nftPageEmptyTitle
+      case .invisible:
+        return Strings.Wallet.nftInvisiblePageEmptyTitle
+      case .spam:
+        return Strings.Wallet.nftSpamPageEmptyTitle
+      }
+    }
+    
+    var emptyDescription: String? {
+      switch self {
+      case .visible:
+        return Strings.Wallet.nftPageEmptyDescription
+      case .invisible, .spam:
+        return nil
+      }
+    }
+  }
+
+  @Published var displayType: NFTDisplayType = .visible
+  @Published private(set) var userVisibleNFTs: [NFTAssetViewModel] = []
+  @Published private(set) var userInvisibleNFTs: [NFTAssetViewModel] = []
```

Comment:
Regarding change from `invisible` to `hidden` comment:
```suggestion
  @Published private(set) var userHiddenNFTs: [NFTAssetViewModel] = []
```